package.json

```diff
@@ -46,11 +47,12 @@
     "modern-normalize": "1.1.0",
     "next": "13.5.4",
     "parse-data-url": "6.0.0",
-    "react": "18.2.0",
-    "react-dom": "18.2.0",
+    "react": "18.3.0-canary-09fbee89d-20231013",
+    "react-dom": "18.3.0-canary-09fbee89d-20231013",
```

Comment:
I had to switch to React `@canary` version in order to use [`React.cache()`](https://react.dev/reference/react/cache). This provides the closest thing we can get to a server-components context. This is also how `next-intl`'s [`unstable_setRequestLocale`](https://next-intl-docs.vercel.app/docs/getting-started/app-router-server-components#add-unstable_setrequestlocale-to-all-layouts-and-pages) works.

Not ideal. Potentially, this can be reverted by always recalculating the env object on demand, as previously. But I wanted to hear your opinion on it first.

Reply:
Reverted back to the stable version.

See:

- https://github.com/prezly/theme-nextjs-bea/pull/929#discussion_r1371523689
- 43c56696b3df051b1232fdf0fe206f5caf1c3bea
- 3eff05e8c7486bccd8269776e3366461ed41108a